src/main/kotlin/org/dif/crypto/key/RecipientKeySelector.kt

```diff
@@ -21,4 +24,55 @@ class RecipientKeySelector(private val didDocResolver: DIDDocResolver, private v
             didDoc.findVerificationMethod(verificationMethodId)
         }
     )
+
+    fun authCryptKeys(fromSelector: String, toSelector: List<String>): Pair<Key, List<Key>> {
+        val (did) = divideDIDFragment(fromSelector)
+        val didDoc = didDocResolver.resolve(did).orElseThrow { throw DIDDocNotResolvedException(did) }
+
+        return if (isDIDFragment(fromSelector)) {
+            val sender = didDocResolver.resolve(did)
```

Comment:
`didDocResolver.resolve(did)` - it's already called above, no need to call potentially expensive operation twice.